src/pybind/mgr/cephadm/serve.py

```diff
@@ -1459,6 +1460,11 @@ async def _run_cephadm(self,
         if not self.mgr.cgroups_split:
             final_args += ['--no-cgroups-split']
 
+        if not timeout:
+            # 15 minute global timeout if no timeout was passed
+            timeout = self.mgr.default_cephadm_command_timeout
+        final_args += ['--timeout', str(timeout)]
+
```

Comment:
I'm a bit iffy on the idea that the timeout passed to cephadm (binary) is the same timeout used in the mgr module. I'd prefer there to be a small "grace period" to allow the opportunity for the lower layer to exit and return and have it's result(s) be processed rather than "simultaneously" terminating the command and giving up on it.  This grace period could be only a few seconds. To have fewer touch points in the code this could just subtract, say 5, from the timeout before passing that adjusted timeout to cephadm.

Reply:
I think that makes sense and this was an oversight while I was focused on just the asyncio timeout. I added the -5 here and also put a 60 second lower bound on the default while I was looking at it (although that's arguably too low, but better than nothing)